packages/template-retail-react-app/app/hooks/use-auth-modal.test.js

```diff
@@ -155,17 +171,63 @@ test('Renders check email modal on email mode', async () => {
     mockUseForm.mockRestore()
 })
 
-test('Renders passwordless login when enabled', async () => {
-    const user = userEvent.setup()
+describe('Passwordless enabled', () => {
+    test('Renders passwordless login when enabled', async () => {
+        const user = userEvent.setup()
 
-    renderWithProviders(<MockedComponent isPasswordlessEnabled={true} />)
+        renderWithProviders(<MockedComponent isPasswordlessEnabled={true} />)
 
-    // open the modal
-    const trigger = screen.getByText(/open modal/i)
-    await user.click(trigger)
+        // open the modal
+        const trigger = screen.getByText(/open modal/i)
+        await user.click(trigger)
 
-    await waitFor(() => {
-        expect(screen.getByText(/continue securely/i)).toBeInTheDocument()
+        await waitFor(() => {
+            expect(screen.getByText(/continue securely/i)).toBeInTheDocument()
+        })
+    })
+
+    test('Allows passwordless login', async () => {
+        const {user} = renderWithProviders(<MockedComponent isPasswordlessEnabled={true} />)
+        const validEmail = 'test@salesforce.com'
+
+        // open the modal
+        const trigger = screen.getByText(/open modal/i)
+        await user.click(trigger)
+
+        await waitFor(() => {
+            expect(screen.getByText(/continue securely/i)).toBeInTheDocument()
+        })
+
+        // enter a valid email address
+        await user.type(screen.getByLabelText('Email'), validEmail)
+
+        // initiate passwordless login
+        const passwordlessLoginButton = screen.getByText(/continue securely/i)
+        // Click the button twice as the isPasswordlessLoginClicked state doesn't change after the first click
+        await user.click(passwordlessLoginButton)
+        await user.click(passwordlessLoginButton)
+        expect(
+            mockAuthHelperFunctions[AuthHelpers.AuthorizePasswordless].mutateAsync
+        ).toHaveBeenCalledWith({
+            userid: validEmail,
+            callbackURI: 'https://webhook.site/27761b71-50c1-4097-a600-21a3b89a546c?redirectUrl=/'
+        })
+
+        // check that check email modal is open
+        await waitFor(() => {
+            const withinForm = within(screen.getByTestId('sf-form-resend-passwordless-email'))
+            expect(withinForm.getByText(/Check Your Email/i)).toBeInTheDocument()
+            expect(withinForm.getByText(validEmail)).toBeInTheDocument()
+        })
+
+        // resend the email
+        user.click(screen.getByText(/Resend Link/i))
+        expect(
+            mockAuthHelperFunctions[AuthHelpers.AuthorizePasswordless].mutateAsync
+        ).toHaveBeenCalledWith({
+            userid: validEmail,
+            callbackURI: 'https://webhook.site/27761b71-50c1-4097-a600-21a3b89a546c?redirectUrl=/'
```

Comment:
nit: is there a way to add a unit test that verfies when `passwordlessConfigCallback` is a relative path it creates the URL correclty?